### DISCUSSION TITLE
Is global configuration mode not applied to all branch configurations by default?

### DISCUSSION THREAD

**Discussion by Garinoth** [GitTools/GitVersion#3473] [Q&A] [answered]
I have noticed since the update to 6.0.0-beta.2 that the default mode in the global configuration was changed from `mode: ContinuousDelivery` to `mode: ContinuousDeployment`, and also the default modes for pull-request and develop were changed, but no the modes of any other branch. This has changed some interactions with my versioning strategy, although not a problem, because I could easily correct them. 

But it has raised a question for me. I was assuming the `mode` from the global configuration section was the default for all the branches, and then it could also be overwritten in each branch configuration section, but most of the sections already have their own default `mode`, which after the change happens to be different from the one in the global configuration. Is this the intended behaviour? In my particular case this means having to modify most branch configurations to `mode: ContinuousDeployment`, which I'm ok with, but in my head it doesn't seem to be the intended behaviour

**Comment by arturcic** [comment]
@HHobeck

**Comment by HHobeck** [comment, marked as answer]
At this moment I'm not sure if this change is final. But independent of that it is important to understand the meaning of defining a fallback branch configuration value and how this values are applying. The breaking change documentation says:

> Additional `fallback` branch configuration properties have been introduced at the root to define base properties which will be inherit to the branch configurations. That means if no other branch configuration in the inheritance line defines the given property the fallback property applies. Notice that the inheritance tree can be controlled using the increment strategy property in the branch configuration section.

So yes we have changed the mode property on the root layer from `ContinuousDelivery` to `ContinuousDeployment`. Means that you are defining the fallback value for this branches where the mode property has not been set and are not inherited from parent branches. And this is true on the branch with name: `main`, `support` and `develop`

**Reply by Garinoth** [reply]
Nice to know, thanks for the response!

I was interpreting that property as a default rather than a fallback. So it is a better practice to explicitly declare the mode for all the branch configurations if we use a different strategy to gitflow (we use a similar strategy, but we ditch the develop branch and use main/master as the base for development, getting our final versions from the release branch)
